src/stp/PlayEvaluator.cpp

```diff
@@ -0,0 +1,142 @@
+                return evaluation::BallCloseToThemGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallCloseToUs:
+                return evaluation::BallCloseToUsGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallClosestToUs:
+                return evaluation::BallClosestToUsGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallGotShot:
+                return evaluation::BallGotShotGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallIsFree:
+                return evaluation::BallIsFreeGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallMovesSlow:
+                return evaluation::BallMovesSlowGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallOnOurSide:
+                return evaluation::BallOnOurSideGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::BallOnTheirSide:
+                return evaluation::BallOnTheirSideGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::DistanceFromBall:
+                return evaluation::DistanceFromBallGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::FreedomOfRobots:
+                return evaluation::FreedomOfRobotsGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::GoalVisionFromBall:
+                return evaluation::GoalVisionFromBallGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::GoalVision:
+                return evaluation::GoalVisionGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::NoGoalVisionFromBall:
+                return evaluation::NoGoalVisionFromBallGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::WeHaveBall:
+                return evaluation::WeHaveBallGlobalEvaluation().metricCheck(_world, &field);
+            case GlobalEvaluation::WeHaveMajority:
+                return evaluation::WeHaveMajorityGlobalEvaluation().metricCheck(_world, &field);
+            default:
+                RTT_WARNING("Unhandled ScoreEvaluation!");
+                return 0;
+        }
+    }
+
+    void PlayEvaluator::update(world::World* _world) noexcept {
+        this->world = _world;
+        this->field = _world->getField().value();
+    }
+
+    world::World* PlayEvaluator::getWorld() noexcept { return world; }
+
+    void PlayEvaluator::clearGlobalScores() {
+        scoresGlobal.clear();
+    }
+
+    bool PlayEvaluator::checkEvaluation(GlobalEvaluation globalEvaluation, uint8_t cutOff) noexcept { return getGlobalEvaluation(globalEvaluation) >= cutOff; }
+
+    uint8_t PlayEvaluator::calculateScore(std::vector<PlayScoring> scoring){
+        double score = 0;
+        double weight = 0;
+        for (auto i : scoring){
```

Comment:
const auto&; more efficient.

Also, try to prevent using 'i'. I think naming it playscore here is much clearer.

Reply:
Ah yeh, was looking for these after the last pr